code/game/objects/items/sticker.dm

```diff
@@ -1,131 +1,70 @@
-/// parent type for all other stickers. do not spawn directly
+#define MAX_STICKER_COUNT 15
+
 /obj/item/sticker
 	name = "sticker"
-	desc = "A sticker with some strong adhesive on the back, sticks to stuff!"
-	item_flags = NOBLUDGEON | XENOMORPH_HOLDABLE //funny
-	resistance_flags = FLAMMABLE
-	icon = 'icons/obj/toys/stickers.dmi'
-	w_class = WEIGHT_CLASS_TINY
-	throw_range = 3
-	vis_flags = VIS_INHERIT_DIR | VIS_INHERIT_PLANE | VIS_INHERIT_LAYER
-	///If not null, pick an icon_state from this list
-	var/icon_states
-	/// If the sticker should be disincluded from normal sticker boxes.
-	var/contraband = FALSE
+	desc = "it sticks to objects"
 
-/obj/item/sticker/Initialize(mapload)
-	. = ..()
-	if(icon_states)
-		icon_state = pick(icon_states)
-	pixel_y = rand(-3,3)
-	pixel_x = rand(-3,3)
-	AddElement(/datum/element/sticker)
-
-/obj/item/sticker/smile
-	name = "smiley sticker"
-	icon_state = "smile"
-
-/obj/item/sticker/frown
-	name = "frowny sticker"
-	icon_state = "frown"
-
-/obj/item/sticker/left_arrow
-	name = "left arrow sticker"
-	icon_state = "larrow"
-
-/obj/item/sticker/right_arrow
-	name = "right arrow sticker"
-	icon_state = "rarrow"
-
-/obj/item/sticker/star
-	name = "star sticker"
-	icon_state = "star1"
-	icon_states = list("star1","star2")
-
-/obj/item/sticker/heart
-	name = "heart sticker"
-	icon_state = "heart"
-
-/obj/item/sticker/googly
-	name = "googly eye sticker"
-	icon_state = "googly1"
-	icon_states = list("googly1","googly2")
-
-/obj/item/sticker/rev
-	name = "blue R sticker"
-	desc = "A sticker of FUCK THE SYSTEM, the galaxy's premiere hardcore punk band."
-	icon_state = "revhead"
-
-/obj/item/sticker/pslime
-	name = "slime plushie sticker"
-	icon_state = "pslime"
-
-/obj/item/sticker/pliz
-	name = "lizard plushie sticker"
+	icon = 'icons/obj/toys/stickers.dmi'
 	icon_state = "plizard"
 
-/obj/item/sticker/pbee
-	name = "bee plushie sticker"
-	icon_state = "pbee"
+	w_class = WEIGHT_CLASS_TINY
 
-/obj/item/sticker/psnake
-	name = "snake plushie sticker"
-	icon_state = "psnake"
+	throw_range = 3
 
-/obj/item/sticker/robot
-	name = "bot sticker"
-	icon_state = "tile"
-	icon_states = list("tile","medbot","clean")
+	pressure_resistance = 0
 
-/obj/item/sticker/toolbox
-	name = "toolbox sticker"
-	icon_state = "toolbox"
+	resistance_flags = FLAMMABLE
+	max_integrity = 50
+
+/**
+ * What stickers can do?
+ *
+ * - They can be attached to any object.
+ * - They inherit cursor position when attached.
+ * - They are unclickable by mouse, I suppose?
+ * - They can be washed off.
+ * - They can be burnt off.
+ * - They can be attached to the object they collided with.
+ * - They play "attack" animation when attached.
+ *
+ */
+
+/obj/item/sticker/Bump(atom/bumped_atom)
+	if(prob(100))
+		attempt_attach(bumped_atom)
+
+/obj/item/sticker/afterattack(atom/target, mob/user, proximity_flag, click_parameters)
```

Comment:
I recommend using `interact_with_atom` rather than `afterattack`

Reply:
I need `click_parameters`, but they are missing in `interact_with_atom`.

Reply:
Tsk, they should really be passed into it. You could add it as a parameter, or if you're lazy make use of `COMSIG_ITEM_INTERACTING_WITH_ATOM` since it gets the modifiers

Reply:
I've seen that previous implementation had some kind of element. I guess, I should use it too if I subscribe to object's own signals?

Reply:
It's technically fine